.github/workflows/build-and-run.yml

```diff
@@ -29,23 +34,26 @@ jobs:
       matrix:
         os:
           - label: Linux
-            runner: ubuntu-latest
-            ctest_exclude: -E ".*($ALL_EXCLUDE)"
-          - runner: macos-latest
-            label: macOS
+          - label: macOS
+            runner: macos-latest
             ctest_exclude: -E ".*($ALL_EXCLUDE|io\.gdal\.raster|unwrap\.snaphu|background)"
             # io.gdal.raster and unwrap.snaphu tests fail due to known unresolved issues
             # with GDAL memory mapping on macOS.
             # background reader/writer test sometimes fails on macOS due to slow runtimes.
         deps:
           - label: Latest
-            env-file: environment.yml
-            extra-specs: ""
         build-type: [ Release, Debug ]
-
+        include:
+          - os:
+              label: Linux
```

Comment:
Is there are way to default `os.label` to Linux? So that we don't need to specify it explicitly whenever we add a new min-reqs job?

Reply:
Maybe we could just get rid of `os.label` (and `deps.label`). These currently are only used to annotate jobs with a description of their behavior.

Maybe a better approach would be to manually label jobs using a parameter called e.g. `action-label`. Any jobs that aren't manually labelled could fall back to using a label which is some combination of the runner type and build type.

Reply:
I wasn't really able to figure out a good way to have a defaultable action-label, because `env` isn't available when evaluating the job name. But I think the way it is currently might be OK actually, because expanding an empty os.label just evaluates to an empty string. So I dropped the os.label specifier here, and it just uses the deps.label instead, so the job names seem to look OK and will only require specifying either os.label or deps.label for future jobs.